### DISCUSSION TITLE
Android Installation - How to verify it completed

### DISCUSSION THREAD

**Discussion by focher** [tesla-android/issue-tracker#65] [General]
After issuing the last `fastboot reboot` command, the instructions state that it will take a long period of time to complete the install - including multiple reboots. Is there a method to verify the installation has fully completed? I never see the device appear again with `fastboot devices` and when I connect it to the first Pi, it always says its waiting for the virtual screen to start.

Would be good to have some way to verify that Android is fully running.

**Comment by cukal** [comment]
I connect a keyboard & HDMI cable to the pi running android and follow on an external screen what's happening. After a while when the install is done you should get the android boot loader text output followed by the flashing android loading screen finally booting into "setup this phone" (as android thinks it's running on a phone). When I got the "waiting for the virtual screen to start" it basically indicated the HDMI capture card wasn't really working and it turned out I connected the HDMI capture card cable the wrong way. The capture card should have a bright green led when powering on the first pi.

**Reply by focher** [reply]
I seem to have the virtual screen working but am still working on why the touchscreen isn't working. I get CarPlay up on the browser display. One step at a time!

Thanks.
